pkg/devcontainer/config/parse.go

```diff
@@ -67,6 +67,46 @@ func SaveDevContainerJSON(config *DevContainerConfig) error {
 	return nil
 }
 
+func ParseDevContainerJSONFile(jsonFilePath string) (*DevContainerConfig, error) {
+	var err error
+	path, err := filepath.Abs(jsonFilePath)
+	if err != nil {
+		return nil, errors.Wrap(err, "make path absolute")
+	}
+
+	bytes, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	devContainer := &DevContainerConfig{}
+	err = json.Unmarshal(jsonc.ToJSON(bytes), devContainer)
+	if err != nil {
+		return nil, err
+	}
+	devContainer.Origin = path
+	return replaceLegacy(devContainer)
+}
+
+
+func ParseDevContainerUserJSON(config *DevContainerConfig) (*DevContainerConfig, error) {
+	filename := filepath.Base(config.Origin)
+	filename = strings.TrimSuffix(filename, filepath.Ext(filename))
+
+	devContainerUserUserFilename := fmt.Sprintf("%s.user.json", filename)
+	devContainerUserUserFilePath := filepath.Join(filepath.Dir(config.Origin), devContainerUserUserFilename)
+
+	_, err = os.Stat(devContainerUserUserFilePath)
+	if err == nil {
+		userConfig, err := ParseDevContainerJSONFile(devContainerUserUserFilePath)
+		if err != nil {
+			return nil, err
+		}
+		return userConfig, nil
+	}
+	return nil, nil
+}
+
 func ParseDevContainerJSON(folder, relativePath string) (*DevContainerConfig, error) {
```

Comment:
I know this isn't your code but a docblock here to distringuish between ParseDevContainerJSONFile would be great :pray: